main.go

```diff
@@ -75,10 +105,23 @@ func (p *Proxy) Provision(ctx caddy.Context) error {
 	}
 	p.auth.TokenSecret = string(secret)
 
-	p.sites = make(map[string]string)
-	p.sites[p.auth.SiteOneLevel] = p.auth.SiteOne
-	p.sites[p.auth.SiteTwoLevel] = p.auth.SiteTwo
-	p.sites[p.auth.SiteThreeLevel] = p.auth.SiteThree
+	p.sites = make(map[string]AuthSite)
+	p.sites[p.auth.SiteOneLevel], err = NewAuthSite(p.auth.SiteOne)
```

Comment:
This was done mainly because of an [issue](https://github.com/kelseyhightower/envconfig/issues/180) with kelseyhightower/envconfig where he use Split instead of SplitN or Cut. So urls with ports cause errors. Given that the project hasn't been updated in a few years, I'm not holding my breath for that fix.

I could make a custom decoder and use that instead, it probably won't actually be much more work. Given that AuthSite is pretty much a custom decoder, I think it's worth doing.